cmd/cluster/context.go

```diff
@@ -147,3 +178,76 @@ func checkSupportStatus(response *cmv1.LimitedSupportReasonsListResponse) error
 
 	return nil
 }
+
+func (o *statusOptions) getServiceLogs() error {
+	// Create OCM client to talk to cluster API
+	ocmClient := utils.CreateConnection()
+	defer func() {
+		if err := ocmClient.Close(); err != nil {
+			fmt.Printf("Cannot close the ocmClient (possible memory leak): %q", err)
+		}
+	}()
+
+	// Use the OCM client to retrieve clusters
+	clusters := utils.GetClusters(ocmClient, []string{o.clusterID})
+	if len(clusters) != 1 {
+		return fmt.Errorf("GetClusters expected to return 1 cluster, got: %d", len(clusters))
+	}
+	cluster := clusters[0]
+
+	var erroredServiceLogs []sl.ServiceLogShort
+	serviceLogListAllMessagesFlag := false
+
+	// Now get the SLs for the cluster
+	slResponse, err := sendRequest(servicelog.CreateListSLRequest(ocmClient, cluster.ExternalID(), serviceLogListAllMessagesFlag))
+	if err != nil {
+		return err
+	}
```

Comment:
isn't this whole section necessary for the `serviclelog list` command as well? Is it possible to reuse/unify it?

Reply:
Yeah it could be - the main difference between the 2 is that `serviclelog list` gets the response and just dumps it out, so all SLs. I'm filtering the SLs here. The `dump.pretty` adds colour to the `servicelog list` command, but I haven't been able to get it to work for the cluster context cmd.

Reply:
I think hes saying you could have something like:
```
func fetchServiceLogs(string clusterId, bool listAllMessagesFlag) sl.ServiceLogShortList, error {
        // Create OCM client to talk to cluster API
	ocmClient := utils.CreateConnection()
	defer func() {
		if err := ocmClient.Close(); err != nil {
			fmt.Printf("Cannot close the ocmClient (possible memory leak): %q", err)
		}
	}()

        ...

        var serviceLogs sl.ServiceLogShortList
	err = json.Unmarshal(slResponse.Bytes(), &serviceLogs)
	if err != nil {
		fmt.Printf("Failed to unmarshal the SL response %q\n", err)
		return err
	}
        return serviceLogs
}
```

And call that in both places. Each caller can format the output however they want.

Reply:
done